kratos/input_output/vtu_output.h

```diff
@@ -0,0 +1,175 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <variant>
+#include <unordered_map>
+
+// External includes
+
+// Project includes
+#include "includes/io.h"
+#include "includes/model_part.h"
+#include "containers/variable.h"
+#include "containers/flags.h"
+#include "containers/container_expression/container_expression.h"
+
+
+namespace Kratos {
+class KRATOS_API(KRATOS_CORE) VtuOutput : public IO
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    using IndexType = std::size_t;
+
+    using SupportedVariables = std::variant<
+                                    const Variable<int>*,
+                                    const Variable<double>*,
+                                    const Variable<array_1d<double, 3>>*,
+                                    const Variable<array_1d<double, 4>>*,
+                                    const Variable<array_1d<double, 6>>*,
+                                    const Variable<array_1d<double, 9>>*>;
+
+    using SupportedCellContainerExpressions = std::variant<
+                                                ContainerExpression<ModelPart::ConditionsContainerType>::Pointer,
+                                                ContainerExpression<ModelPart::ElementsContainerType>::Pointer>;
+
+    KRATOS_CLASS_POINTER_DEFINITION(VtuOutput);
+
+    KRATOS_DEFINE_LOCAL_FLAG( NODES );
+    KRATOS_DEFINE_LOCAL_FLAG( CONDITIONS );
+    KRATOS_DEFINE_LOCAL_FLAG( ELEMENTS );
+
+    ///@}
+    ///@name Public classes
+    ///@{
+
+    struct VariableComparator
+    {
+        bool operator()(
+            const SupportedVariables& rFirst,
+            const SupportedVariables& rSecond) const
+        {
+            bool is_less = false;
+            std::visit([&is_less](auto pFirst, auto pSecond) {
+                is_less = pFirst->Key() < pSecond->Key();
+            }, rFirst, rSecond);
+            return is_less;
```

Comment:
`std::variant` returns the result of the input lambda.
```suggestion
            return std::visit([](auto pFirst, auto pSecond) {
                return pFirst->Key() < pSecond->Key();
            }, rFirst, rSecond);
```

Reply:
Nice :D...